cdap-examples/Purchase/src/main/java/co/cask/cdap/examples/purchase/PurchaseApp.java

```diff
@@ -69,7 +70,7 @@ public void configure() {
 
     // Schedule the workflow
     schedule(
-      buildSchedule("DailySchedule", "PurchaseHistoryWorkflow")
+      buildSchedule("DailySchedule", SchedulableProgramType.WORKFLOW, "PurchaseHistoryWorkflow")
```

Comment:
I think in general all `ProgramType`s are schedulable (although the current `ProgramType` is not defined very well, as it has `CUSTOM_ACTION` there, which is not a program. We should get rid of that).

In fact, in the current implementation, I see no point why we have to restrict to workflow at all, since we are taking a `ProgramLifecycleService` in `ConstraintCheckerService` to start workflow, we can start any program type already.

For the `SchedulableProgramType`, it actually represents program type that can goes inside a workflow (i.e. MR, Spark, CustomAction), so we should definitely fix it and rename it appropriately.